tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -248,9 +254,21 @@ private void InstrumentationGateway_AddHeadersResponseTags(object sender, Instru
         private void Report(ITransport transport, Span span, string resultData, bool blocked)
         {
             span.SetTag(Tags.AppSecEvent, "true");
-            var samplingPirority = _settings.KeepTraces
-                                       ? SamplingPriority.UserKeep : SamplingPriority.AutoReject;
-            span.SetTraceSamplingPriority(samplingPirority);
+            var exceededTraces = _rateLimiter.UpdateTracesCounter();
+            if (exceededTraces == 0)
+            {
+                // NOTE: DD_APPSEC_KEEP_TRACES=false means "drop all traces by setting AutoReject".
+                // It does _not_ mean "stop setting UserKeep (do nothing)". It should only be used for testing.
+                span.SetTraceSamplingPriority(_settings.KeepTraces ? SamplingPriority.UserKeep : SamplingPriority.AutoReject);
+            }
+            else
+            {
+                span.SetMetric(Metrics.AppSecRateLimitDroppedTraces, exceededTraces);
```

Comment:
Ok let's be honest, I'just discovered this metric thingy and I don't know how it works. So please ignore If my comment is stupid (or better, please someone explain me :p).

Though, here, as we attach a metric to a span. I would have guessed that we need to just set this to 1. And that the aggregation would be done on the backend? No? (in which case you don't need the _exceedTraces in the rate limiter btw)

Reply:
You raise a good question. The RFC says

> The number of AppSec traces ignored by the AppSec rate limiter should be counted by the counter metric: `_dd.<library prefix>.appsec.rate_limit.dropped_traces`

I would have thought that means dogstatsd, not span "metrics" (which are really just numeric tags with a bad name).